shared/src/Datadog.Trace.ClrProfiler.Native/workload_selection_impl.cpp

```diff
@@ -0,0 +1,147 @@
+#include "workload_selection_impl.h"
+
+#include "../../../shared/src/native-src/string.h"
+#include "log.h"
+#include "util.h"
+
+#include <optional>
+extern "C"
+{
+#include <dd/policies/error_codes.h>
+#include <stdio.h>
+    void _except_handler4_common(void)
+    {
+    }
+}
+
+namespace datadog::shared::nativeloader
+{
+namespace
+{
+    enum class InjectionStatus : uint8_t
+    {
+        ALLOW,
+        DENY,
+        UNKNOWN
+    };
+
+    InjectionStatus injection_status;
+
+    std::optional<plcs_errors> evaluatePolicies()
+    {
+        const auto policies_file = GetPoliciesPath();
+
+        FILE* file = NULL;
+        fopen_s(&file, policies_file.string().c_str(), "rb");
+        if (!file)
+        {
+            return PLCS_ENO_DATA;
+        }
+
+        fseek(file, 0, SEEK_END);
+        const auto file_size = ftell(file);
+        fseek(file, 0, SEEK_SET);
+
+        std::vector<uint8_t> buffer(file_size);
+
+        const auto read_size = fread(buffer.data(), 1, file_size, file);
+        fclose(file);
+
+        if (read_size != file_size)
+        {
+            return PLCS_ENO_DATA;
+        }
+
+        auto res = plcs_evaluate_buffer(buffer.data(), buffer.size());
+        if (res != PLCS_ESUCCESS)
+        {
+            return res;
+        }
+
+        return std::nullopt;
+    }
+
+    plcs_errors onInjectionAllow(plcs_evaluation_result eval_result, char**, size_t, const char* policy, int)
+    {
+        if (injection_status != InjectionStatus::UNKNOWN) return PLCS_ESUCCESS;
+
+        injection_status = (eval_result == PLCS_EVAL_RESULT_TRUE) ? InjectionStatus::ALLOW : InjectionStatus::UNKNOWN;
+
+        return PLCS_ESUCCESS;
+    }
+
+    plcs_errors onInjectionDeny(plcs_evaluation_result eval_result, char**, size_t, const char* policy, int)
+    {
+        if (injection_status != InjectionStatus::UNKNOWN) return PLCS_ESUCCESS;
+
+        injection_status = (eval_result == PLCS_EVAL_RESULT_TRUE) ? InjectionStatus::DENY : InjectionStatus::UNKNOWN;
+
+        return PLCS_ESUCCESS;
+    }
+
+    std::string GetDotnetDll()
+    {
+        const auto [_, tokenized_command_line] = ::shared::GetCurrentProcessCommandLine();
+        if (tokenized_command_line.size() < 2)
+        {
+            return "";
+        }
+
+        if (tokenized_command_line[0] != L"dotnet" && tokenized_command_line[0] != L"dotnet.exe")
+        {
+            return "";
+        }
+
+        if (tokenized_command_line[1].empty() || !tokenized_command_line[1].ends_with(L".dll"))
+        {
+            return "";
+        }
+
+        auto dll_path = fs::path(::shared::ToString(tokenized_command_line[1]));
+        return dll_path.filename().string();
+    }
+
+} // namespace
+
+bool wls()
+{
+    const auto process_name = ::shared::ToString(::shared::GetCurrentProcessName());
```

Comment:
nit: we read this already in cor_profiler, we can probably just pass it in?

Reply:
addressed in 813b52d67546743034aa3c60f731503d9a44291e